File: backend/src/main/java/com/tripfriend/domain/review/dto/ReviewResponseDto.java
package com.tripfriend.domain.review.dto;

import com.tripfriend.domain.review.entity.Review;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
public class ReviewResponseDto {
    private Long reviewId;
    private String title;
    private String content;
    private double rating;
    private Long memberId;
    private String memberName;
    private String profileImage;
    private Long placeId;
    private String placeName;
    private int viewCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private int commentCount;

    public ReviewResponseDto(Review review, String memberName, int commentCount) {
        this.reviewId = review.getReviewId();
        this.title = review.getTitle();
        this.content = review.getContent();
        this.rating = review.getRating();
        this.memberId = review.getMember().getId();
        this.memberName = memberName;
        this.profileImage = null;
        this.placeId = review.getPlace().getId();
        this.placeName = review.getPlace().getPlaceName();
        this.viewCount = 0;
        this.createdAt = review.getCreatedAt();
        this.updatedAt = review.getUpdatedAt();
        this.commentCount = commentCount;
    }

    public ReviewResponseDto(Review review, String memberName, int commentCount, String profileImage) {
        this(review, memberName, commentCount);
        this.profileImage = profileImage;
    }

    public void setViewCount(int viewCount) {
        this.viewCount = viewCount;
    }

}

File: backend/src/main/java/com/tripfriend/domain/review/dto/ReviewResponseDto.kt
package com.tripfriend.domain.review.dto

import com.tripfriend.domain.review.entity.Review
import java.time.LocalDateTime

data class ReviewResponseDto(
    val reviewId: Long? = null,
    val title: String = "",
    val content: String = "",
    val rating: Double = 0.0,
    val memberId: Long? = null,
    val memberName: String = "",
    var profileImage: String? = null, // var로 변경하여 수정 가능하게 함
    val placeId: Long? = null,
    val placeName: String? = null,
    var viewCount: Int = 0,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
    val commentCount: Int = 0
) {
    constructor(review: Review, memberName: String, commentCount: Int) : this(
        reviewId = review.reviewId,
        title = review.title,
        content = review.content,
        rating = review.rating,
        memberId = review.member?.id,
        memberName = memberName,
        profileImage = null,
        placeId = review.place?.id,
        placeName = review.place?.placeName,
        viewCount = 0,
        createdAt = review.createdAt,
        updatedAt = review.updatedAt,
        commentCount = commentCount
    )

    constructor(review: Review, memberName: String, commentCount: Int, profileImage: String?) : this(
        reviewId = review.reviewId,
        title = review.title,
        content = review.content,
        rating = review.rating,
        memberId = review.member?.id,
        memberName = memberName,
        profileImage = profileImage,
        placeId = review.place?.id,
        placeName = review.place?.placeName,
        viewCount = 0,
        createdAt = review.createdAt,
        updatedAt = review.updatedAt,
        commentCount = commentCount
    )
}